pkg/metrics/metrics.go

```diff
@@ -31,6 +31,11 @@ const (
 	// ErrorProxy is the value used to notify errors on Proxy.
 	ErrorProxy = "proxy"
 
+	// HistogramFactor configures the factor used for exponential/native histogram in Prometheus client
+	HistogramFactor = 1.1
```

Comment:
This is better documented here: 

https://github.com/prometheus/client_golang/blob/main/prometheus/histogram.go#L398-L429

> The smaller
> // NativeHistogramBucketFactor, the more buckets will be used and thus
> 	// the more costly the histogram will become. A generally good trade-off
> 	// between cost and accuracy is a value of 1.1 (each bucket is at most
> 	// 10% wider than the previous one), which will result in each power of
> 	// two divided into 8 buckets (e.g. there will be 8 buckets between 1
> 	// and 2, same as between 2 and 4, and 4 and 8, etc.).
> 	//
> 	// Details about the actually used factor: The factor is calculated as
> 	// 2^(2^n), where n is an integer number between (and including) -8 and
> 	// 4. n is chosen so that the resulting factor is the largest that is
> 	// still smaller or equal to NativeHistogramBucketFactor. Note that the
> 	// smallest possible factor is therefore approx. 1.00271 (i.e. 2^(2^-8)
> 	// ). If NativeHistogramBucketFactor is greater than 1 but smaller than
> 	// 2^(2^-8), then the actually used factor is still 2^(2^-8) even though
> 	// it is larger than the provided NativeHistogramBucketFactor.

Reply:
The choice of these two parameters should depend on the range of expected values for each metric, otherwise you end up with a histogram that is "tail-heavy".

I think at the least, each histogram should have this configurable, using these values as defaults.

The choice of these metrics is fairly significant, because once we started reporting these metrics with certain bucket sizes we can't change them without affecting the meaning of the metric for all users. We should document typical ranges for each of these metrics, and how these choices have influenced these bucket size metrics.

The final bucket range with these values appears to be 1.1^128 ~200000x the minimum value, based on my cursory understanding of this feature.